store/file/ods.go

```diff
@@ -130,6 +130,31 @@ func writeAxisRoots(w io.Writer, roots *share.AxisRoots) error {
 	return nil
 }
 
+// CheckODSSize checks if the file under given FS path has the expected size.
+func CheckODSSize(path string, eds *rsmt2d.ExtendedDataSquare) error {
+	_, err := checkODSSize(path, eds)
+	return err
+}
+
+func checkODSSize(path string, eds *rsmt2d.ExtendedDataSquare) (*headerV0, error) {
+	ods, err := OpenODS(path)
+	if err != nil {
+		return nil, fmt.Errorf("opening ODS file: %w", err)
+	}
+
+	shares := filledSharesAmount(eds)
+	expectedSize := ods.hdr.OffsetWithRoots() + shares*ods.hdr.ShareSize()
```

Comment:
Shouldnt we verify header as well instead of trusting it? Whats if header also corrupted?

Reply:
It will fail to read the header in check and return the error (same results as corrupted)

Reply:
If it has corrupted values it is very unlikely it will match exactly expected file size, and will fail the check. Keep in mind this is only Size check, not a full content inspection. In case of full content integrity check it would be necessary to verify headers values as well as content of roots and data.

Reply:
> If the header has corrupted values it is very unlikely it will match exactly expected file size

This is what concerns me, we don't know if it will match or not. I don't think trusting header is valid when we dont trust the file. We could simply get the file size from the eds, arent we?

Reply:
Share size is tricker than length, but I think I have a workaround. We can use the size of first share. So we don't have to trust header on disk at all. Will update the code